### DISCUSSION TITLE
Merge two adjacent BinaryRegions

### DISCUSSION THREAD

**Discussion by tomasmanu** [bonsai-rx/bonsai#1031] [Q&A] [answered]
Hi there!
   I'd like to merge two binary regions into only one binary region (the reason why I want that is that the two regions correspond to the same object which gets divided because of another object that covers it in the setup).
![image](https://user-images.githubusercontent.com/69548554/189261641-b634af9f-169c-4b04-bafc-e78521b69fa8.png)
I found a MergeBinaryRegions node and an AverageBinaryRegion node but I couldn't find the way to use them. My workflow looks like this:
![image](https://user-images.githubusercontent.com/69548554/189262656-0d29d021-7c46-47fc-8165-d25684218e19.png)

My first thought was to obtain the corners of each BinaryRegion so that I can get the big binary region by adding the coordinates of the corners, but I don't know if it's possible. Because of that I thought of the MergeBinaryRegions and the AverageBinaryRegion nodes. I would really appreciate it if you suggest how can I proceed to merge those regions. I want to obtain the big BinaryRegion because I want to calculate its centroid and then use it to calculate Heading direction.
Any help would be appreciated.
Cheers,
Tomás

**Comment by bruno-f-cruz** [comment, marked as answer]
Hey Tomas!

`MergeBinaryRegions` is used to merge two collections of ConnectedComponents, not to merge single binary regions. It takes a Tuple of two `ConnectedComponentCollections `.

`AverageBinaryRegions` might work given what you are describing, but you should add it right after SortBinaryRegions. After you use `Concat` you are effectively flattening the Collection into a sequence of its single elements. 

Hope it helps,
Bruno

**Reply by tomasmanu** [reply]
Hi Bruno!
 I tried using `AverageBinaryRegions` after `SortBinaryRegions` but nothing gets displayed in the first one:

![image](https://user-images.githubusercontent.com/69548554/189369258-ad12402b-ac49-4572-a97f-ccc1a07a3e71.png)

How can I make it work?
I leave attached the workflow and the video I use
[workflow_video.zip](https://github.com/bonsai-rx/bonsai/files/9536112/workflow_video.zip)

**Reply by bruno-f-cruz** [reply]
I don't think the visualizer will work since there are a few missing fields. However the centroid should be correct.
